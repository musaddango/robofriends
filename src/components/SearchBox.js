import React from 'react';

const SearchBox = ({ searchChange }) => {
  console.log('SearchBox')
  return (
    <div className='pa2'>
      <input
        name='searchBox'
        className='pa3 ba b--green bg-lightest-blue'
        type='search'
        placeholder='search robots'
        onChange={searchChange}
      />
    </div>
  );
}

export default SearchBox;