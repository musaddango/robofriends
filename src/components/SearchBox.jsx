import 'tachyons';

// Searchbox component

const SearchBox = ({ searchChange }) => {
	return(
		<div className='pa2'>
			<input 
			type='search' 
			className='pa3 ba b--green bg-lightest-blue'
			placeholder='Search Robots'
			onChange= {searchChange}
			/>
		</div>
	);
}

export default SearchBox;