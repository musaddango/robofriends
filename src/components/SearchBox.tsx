import React from 'react';

interface ISearchProps {
  searchChange: React.ChangeEventHandler<HTMLInputElement> | undefined,
}

const SearchBox: React.FC<ISearchProps> = ({ searchChange }) => {
  return (
    <div className='pa2'>
      <input
        name='searchBox'
        className='pa3 ba b--green bg-lightest-blue'
        type='search'
        placeholder='search robots'
        onChange={searchChange}
      />
    </div>
  );
}

export default SearchBox;