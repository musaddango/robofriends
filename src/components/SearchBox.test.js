import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SearchBox from './SearchBox';

it('expect to render Card component', ()=>{
    const wrapper = shallow(<SearchBox />)

    expect(toJson(wrapper)).toMatchSnapshot();
});