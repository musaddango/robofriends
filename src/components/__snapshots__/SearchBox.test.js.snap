// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render Card component 1`] = `
<div
  className="pa2"
>
  <input
    className="pa3 ba b--green bg-lightest-blue"
    name="searchBox"
    placeholder="search robots"
    type="search"
  />
</div>
`;
